### DISCUSSION TITLE
Motivação

### DISCUSSION THREAD

**Discussion by Adejair** [frontendbr/forum#639] [Discussão]
Olá queridos devs, 

Eu gostaria de abrir uma discussão sobre o titulo motivação, vocês já se sentiram muito desmotivados no serviço de vocês, ver uma coisa que vocês amam fazer e  simplismente  passarem a odiar. Este é o meu caso, eu passei a odiar qualquer tipo de código, tem uns 5 meses, meu trabalho não está rendendo nada mais. 

Estou fazendo bacharel em Engenharia de Software e estou fazendo estagio em desenvolvimento, na maior parte do tempo no desenvolvimento front e back. Simplismente estou sem motivação para fazer nada, vocês já passaram por isso? como superaram?

**Comment by kvnol** [comment]
@Adejair cara, todos nós passamos por isso, ainda mais no início de nossas carreiras quando chegamos à mesmície do dia-a-dia e não vemos evolução. E ainda mais quando queremos ser como grandes nomes do desenvolvimento (posso citar vários nomes aqui), que desenvolvem coisas incríveis e a galera elogia muito os trampos.

Uma saída que tive disso, primeiramente, foi **participar de fóruns** (como esse, olha só :clap:), me engajar com a galera (como amo design, estou participando da construção da marca do [Front-end BR](https://github.com/frontendbr/sugestoes/issues/20) e da [Open School Brasil](https://github.com/openschoolbr/forum/issues/3#issuecomment-320000992)) e fazer amizade com eles. O segundo ponto é **estar presente em eventos** como o FrontinSampa e meetups, você conversa com todo tipo de gente (de iniciante a avançado), faz networking e sai de lá inspirado a aprender coisas novas e meter a cara no mundo.

Essa foi a saída que tive para sair da mesmície, tente também. Abraços! :smile:

**Comment by WandersonAlves** [comment]
No meu caso foi diferente.

Trabalhava a 3 anos em uma empresa, com vários projetos e eu sempre tava com algo "diferente". Porém nos últimos dois anos, comecei a trabalhar com Angular. Foi muito bom inclusive, mas o que me deixou desmotivado não foi codificar em si, foi as pessoas, os processos, a falta de organização, e isso se refletia em código...manutenção em código ruim, ilegível e essas coisas mais que todo mundo sempre sabe.

Eu amo programar, mas nesses dois últimos anos, programar em horário de trabalho era insuportável.
E então cheguei ao ponto de ir procurar novas oportunidades e procurar principalmente **mais tempo para fazer minhas codificações.**

Hoje me encontro num lugar que consigo conciliar minhas atividades pessoais (desenvolver em VueJS por exemplo) e continuar as 8hrs trabalhando com o que fui contratado (AngularJS).

Dê um tempo pra si e procure coisas novas, sempre tem um lugar que suporte a gente.

**Comment by guilhermewaess** [comment]
Acredito que meu caso é bem parecido com o do @WandersonAlves e acho que se alguém ainda não passou, vai passar pelo processo de ficar desanimado não tanto pelo código, mas sim por consequências causadas por pessoas/processos. Depois de um tempo você vai acabar aprendendo a lidar com esse tipo de situação. Eu tb consigo conciliar meu trabalho com as atividades pessoais, e isso ajuda bastante, o importante é sempre manter o foco em algo que você sente prazer em fazer. Sugiro ainda ler alguns artigos sobre a crise do impostor, muita gente passa por isso e nem sabe.

**Comment by felipefialho** [comment]
@Adejair Escrevi esses dias, da uma olhada > [A motivação desmotivacional no mundo do Desenvolvimento](https://medium.com/@lfeh/a-motiva%C3%A7%C3%A3o-desmotivacional-no-mundo-do-desenvolvimento-d0a9fcd47fd1)

**Comment by ericdouglas** [comment]
@Adejair você nos disse a consequência:

> "...passei a odiar qualquer tipo de código, tem uns 5 meses, meu trabalho não está rendendo nada mais".

Saberia dizer o que **causou** esse sentimento? O que mudou 5 meses atrás que refletiu nisso?

Recomendo fortemente pensar no que está **causando** esse sentimento, só assim você vai conseguir de fato achar uma solução.

Se souber a causa e compartilhar conosco, será mais fácil achar depoimentos que se encaixem com o que está passando, sendo mais efetiva a colaboração.

Parabéns pela coragem de compartilhar isso, e caso tenha possibilidade talvez seja uma boa buscar uma ajuda profissional (psicólogo). Não descuide da sua saúde mental/emocional.

Se precisar de algo estamos ai, meu e-mail está no perfil aqui do GH.

**Comment by khaosdoctor** [comment]
É comum você se sentir desmotivado, principalmente com a quantidade de coisas que existem pra aprender, quando você vê o que acontece e tudo que você precisa saber para entrar no mundo do desenvolvimento, é normal se desmotivar um pouco.

O que me ajudou muito mesmo foi conhecer o pessoal das comunidades como o [ABC-Dev](https://github.com/abc-dev) e o [Training Center](trainingcenter.io), que são comunidades daqui de onde eu moro. Conhecer pessoas, ir em meetups e conversar sobre o assunto vai fazer você entender outros pontos de vista, sem falar do networking que você vai conseguir que pode te ajudar muito mesmo tanto pessoalmente quanto profissionalmente. Dá uma olhada no projeto da [Mentoria](https://github.com/training-center/mentoria) que pode ser uma boa ajuda pra você :)

**Comment by tassoevan** [comment]
Dado que todos já falaram sobre a estagnação sendo uma causa, talvez a adição sobre saúde mental seja importante também (embora eu acredite que o tempo citado seja relativamente curto para ter se desenvolvido um problema de saúde). A área de TI é reconhecida como uma das mais mentalmente desgastantes e existem várias organizações mundo afora promovendo que as empresas tenham mais responsabilidade ao identificar depressão, ansiedade etc. como doenças ocupacionais. Especificamente sobre o relatado, a anedonia é um dos sintomas mais significativos da depressão.

**Comment by barraponto** [comment]
Com certeza, ter acesso a uma comunidade ajuda muito.
No meu caso, eu às vezes tenho essa exaustão e largo o emprego.
É recorrente e complicado -- especialmente quando se tem família pra sustentar e tal.
Pra além de encontrar uma boa relação de trabalho (de tempo, de renda, de afinidade), é valioso cuidar da própria saúde mental.

**Comment by guiajlopes** [comment]
Tenho esse problema também de algumas vezes me desanimar com o trabalho, normalmente sinto isso por falta de aprendizagem técnica no ambiente onde estou. Uma dica que me ajuda é procurar coisas por si só e ir a eventos de tecnologia. Contribuir para comunidade também sempre é bom, ainda mais quando você vê outras pessoas trabalhando em seus projetos e usando, é muito recompensador

**Comment by Sergioamjr** [comment]
Mude alguma coisa, pode ser sua rotina, trabalho, stacks, IDE's. Eu passei por algo parecido quando estava na minha agência, não aguentava mais mexer com PHP, Wordpress, CSS. Pedi demissão, estudei bastante o que eu queria (JS) e há um mês comecei a trabalhar numa startup e voltei a ficar animado com o que faço. Claro que as vezes quero chorar, acho que não consigo dar conta de algumas coisas novas, mas vou resolvendo uma task de cada vez e aprendendo mais.

**Comment by woliveiras** [comment]
@Adejair uma dúvida que precisamos questionar e muitas vezes esquecemos: você já informou essa desmotivação e entristecimento para a sua liderança ou gestão?

Muitas vezes essa galera pode te passar trabalhos mais desafiadores ou menos complexos dependendo do motivo pelo qual você está desmotivado.

São muitos os fatores que podem levar a desmotivação, as vezes as nossas dicas aqui podem te ajudar, mas as vezes somente uma conversa com sua gestão é que vai fazer a mudança no seu dia a dia no trabalho, viu.

Em todo caso estamos aqui pra ajudar sempre que se sentir desmotivado. Entra no Slack do frontendbr pra discutir em tempo real sempre que estiver se sentindo mal. Isso pode te ajudar a ter uma injeção de ânimo num momento ruim.

O Training Center também está ai pra isso, como disse o @khaosdoctor.

**Comment by jmurowaniecki** [comment]
@woliveiras 
> @Adejair uma dúvida que precisamos questionar e muitas vezes esquecemos: você já informou essa desmotivação e entristecimento para a sua liderança ou gestão?

Se a liderança/gestão dele não percebeu (ou ignorou), honestamente é porque **pouco se importa.**

Infelizmente nessa nossa área é assim que as coisas funcionam: não passamos de engrenagens de um grande sistema, onde é muito mais fácil trocar uma engrenagem que dá problema por outra do que *investir em reparo*.

E aquela boa e velha máxima que diz **"ninguém é insubstituível"** se aplica como uma luva **sempre**.

O mercado é controlado por baixo e a regra é o maior rendimento pelo menor salário - se pararmos pra analisar essa questão veremos que não é apenas *o nosso mercado* que é assim (muitas vezes nós mesmos nos confrontamos com esses dilemas quando precisamos comprar alguma coisa ou contratar algum serviço - qual operadora oferece o melhor pacote pelo menor custo?).

---

Em resumo, só gostaria de deixar aqui expressa *minha insatisfação pra com o mercado* - mesmo sabendo que não mudaremos ele nos próximos vinte anos (ao menos não pra melhor).

No entanto te garanto que tu ainda vais encontrar equipes com as quais vais querer trabalhar - e a motivação vai vir dessa conexão com o time (e não do código em si).

Sempre vai cair no teu colo uma granada de mão sem pino e a verdade é que alguém vai ter de desarmar ela em segurança - muitas vezes, se caiu no teu colo, esse alguém já está definido :smile: :hankey: 

Então @Adejair, não desanima com besteira.

Dá o teu melhor. Luta pra que os verdadeiros padrões sejam mantidos. Para que os comentários (quando necessários) sejam realizados, e que o sagrado coding standart seja perpetuado.

---

:candle: Vou acender uma vela aqui para iluminar teus branches. Vai em #pás bro.

_P.S.: força que melhora._

**Comment by woliveiras** [comment]
> Infelizmente nessa nossa área é assim que as coisas funcionam: não passamos de engrenagens de um grande sistema, onde é muito mais fácil trocar uma engrenagem que dá problema por outra do que investir em reparo.

> O mercado é controlado por baixo e a regra é o maior rendimento pelo menor salário

Mas isso é em toda área,  mano. É o "mercado".

Mas o que digo pra questionar é por que as vezes a pessoa realmente não demonstra descontentamento. Dificilmente minha liderança vai me ver desmotivado por que eu gosto de zoar, de brincar e quando estou trabalhando é do mesmo jeito. Eu tento não transpassar para outras pessoas da equipe pra que as demais não fiquem desmotivadas por tabela.

Ai, quando estou chateado com algo eu vou lá e comento. 

> No entanto te garanto que tu ainda vais encontrar equipes com as quais vais querer trabalhar - e a motivação vai vir dessa conexão com o time (e não do código em si).

Concordo!
